### PR TITLE
Respect variables with no value provided

### PR DESCRIPTION
**Pull request by guymers**
When resolving a value for a variable, if a value has not been provided
and it has no default then do not include it instead of setting it to
`null`.

### CODE REVIEW

File: core/src/main/scala/caliban/execution/Field.scala [hunk 1]
```diff
@@ -97,20 +97,21 @@ object Field {
...
+        case InputValue.ListValue(values)   =>
+          Some(InputValue.ListValue(values.flatMap(resolveVariable)))
         case InputValue.ObjectValue(fields) =>
-          InputValue.ObjectValue(fields.map { case (k, v) => k -> resolveVariable(v) })
+          Some(InputValue.ObjectValue(fields.flatMap { case (k, v) => resolveVariable(v).map(k -> _) }))
         case InputValue.VariableValue(name) =>
-          (for {
-            definition  <- variableDefinitions.find(_.name == name)
-            defaultValue = definition.defaultValue getOrElse NullValue
-            value        = variableValues.getOrElse(name, defaultValue)
```
Comment (frekw) [inline core/src/main/scala/caliban/execution/Field.scala:109]: nit 💅 : `getOrElse`

File: core/src/main/scala/caliban/execution/Field.scala [hunk 2]
```diff
@@ -97,20 +97,21 @@ object Field {
...
-          } yield value) getOrElse NullValue
-        case value: Value                   => value
+          for {
+            definition <- variableDefinitions.find(_.name == name)
+            value      <- variableValues.get(name).orElse(definition.defaultValue)
```
Comment (frekw) [inline core/src/main/scala/caliban/execution/Field.scala:109]: nit 💅 : getOrElse

**Review by frekw — APPROVED** [review]
LGTM! ⭐

File: core/src/main/scala/caliban/execution/Field.scala (hunk 2 above)
Comment (guymers) [reply core/src/main/scala/caliban/execution/Field.scala:109]: `definition.defaultValue` is an `Option` so `getOrElse` is not applicable?

Comment (ghostdogpr) [reply core/src/main/scala/caliban/execution/Field.scala:109]: Yeah I think your right

**Comment by ghostdogpr** [comment]
Thanks!

File: core/src/main/scala/caliban/execution/Field.scala (hunk 2 above)
Comment (frekw) [reply core/src/main/scala/caliban/execution/Field.scala:109]: But so is `variableValues.get(name)` since you're getting it from a map, right?

Comment (ghostdogpr) [reply core/src/main/scala/caliban/execution/Field.scala:109]: @frekw `getOrElse` expects an `A`, but `defaultValue` is an `Option[A]`. It's `orElse`, not `getOrElse`.

Comment (frekw) [reply core/src/main/scala/caliban/execution/Field.scala:109]: I meant `getOrElse` from Map :) https://scastie.scala-lang.org/jywLcoZARNG8KOrI7LDHqw

Comment (frekw) [reply core/src/main/scala/caliban/execution/Field.scala:109]: But in either case, doesn't matter much.

Comment (ghostdogpr) [reply core/src/main/scala/caliban/execution/Field.scala:109]: No, `b` is of type `Any` if you do that :P

https://scastie.scala-lang.org/YjZ63y0XRXSxAMGDC6K65Q

Comment (frekw) [reply core/src/main/scala/caliban/execution/Field.scala:109]: 😬

Comment (frekw) [reply core/src/main/scala/caliban/execution/Field.scala:109]: And also 🤦  I guess

**Pull request merged**